driver-core/src/main/java/com/datastax/driver/core/policies/DefaultRetryPolicy.java

```diff
@@ -135,4 +135,17 @@ public RetryDecision onUnavailable(Statement statement, ConsistencyLevel cl, int
             ? RetryDecision.tryNextHost(cl)
             : RetryDecision.rethrow();
     }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation triggers a retry on the next host in the query plan,
+     * regardless of the consistency level, the number of retries or
+     * the possibility that the mutation has been applied server-side.
+     */
+    @Override
+    public RetryDecision onUnexpectedError(Statement statement, ConsistencyLevel cl, int nbRetry, boolean mightHaveBeenApplied) {
```

Comment:
Yes, too much information, and also the fact that some of the exceptions are not public so we would need to change that or rewrap.
On a side note, I'm not a huge fan of the method name, all errors are kind of unexpected; how about `onRequestError` instead?